packages/components/src/components/button/button.tsx

```diff
@@ -1,11 +1,41 @@
 import classNames from 'classnames';
-import PropTypes from 'prop-types';
-import React from 'react';
-import ButtonLoading from './button_loading.jsx';
+import React, { HTMLProps, ReactNode } from 'react';
+import ButtonLoading from './button_loading';
 import Icon from '../icon';
 import Text from '../text';
 
-const ButtonGroup = ({ children, className }) => (
+type TButtonGroupProps = HTMLProps<HTMLDivElement>;
+
+type TButtonProps = HTMLProps<HTMLButtonElement> & {
+    alternate: boolean;
+    blue: boolean;
+    classNameSpan: string;
+    green: boolean;
+    has_effect: boolean;
+    icon: ReactNode;
+    is_button_toggle: boolean;
+    is_circular: boolean;
+    is_circle: boolean;
+    is_disabled: boolean;
+    is_loading: boolean;
+    is_plus: boolean;
+    is_submit_success: boolean;
+    large: boolean;
+    medium: boolean;
+    primary: boolean;
+    primary_light: boolean;
+    renderText: (param: ReactNode) => ReactNode;
+    rounded: boolean;
+    secondary: boolean;
+    small: boolean;
+    tertiary: boolean;
+    text: string;
+    transparent: boolean;
+    type: 'button' | 'submit' | 'reset';
+    wrapperClassName: string;
+};
```

Comment:
```suggestion
type TButtonProps = React.PropsWithChildren<React.ButtonHTMLAttributes<HTMLButtonElement>> & {
    alternate: boolean;
    blue: boolean;
    classNameSpan: string;
    green: boolean;
    has_effect: boolean;
    icon: ReactNode;
    is_button_toggle: boolean;
    is_circular: boolean;
    is_circle: boolean;
    is_disabled: boolean;
    is_loading: boolean;
    is_plus: boolean;
    is_submit_success: boolean;
    large: boolean;
    medium: boolean;
    primary: boolean;
    primary_light: boolean;
    renderText: (param: ReactNode) => ReactNode;
    rounded: boolean;
    secondary: boolean;
    small: boolean;
    tertiary: boolean;
    text: string;
    transparent: boolean;
    wrapperClassName: string;
};
```
React.ButtonHTMLAttributes<HTMLButtonElement> will help automatically infer a native button's type🙂 @jim-deriv